src/envoy/mixer/mixer_control.cc

```diff
@@ -382,6 +388,32 @@ void MixerControl::BuildTcpReport(
       Attributes::TimeValue(std::chrono::system_clock::now());
 }
 
+// Mesh attributes from Pilot are all string type.
+// The attributes with ".ip" suffix will be treated
+// as ipv4 and use BYTES attribute type.
+void MixerControl::SetMeshAttribute(const std::string& name,
+                                    const std::string& value,
+                                    Attributes* attr) const {
+  // Check with ".ip" suffix,
+  if (name.length() <= kIPSuffix.length() ||
+      name.compare(name.length() - kIPSuffix.length(), kIPSuffix.length(),
+                   kIPSuffix) != 0) {
+    attr->attributes[name] = Attributes::StringValue(value);
+    return;
+  }
+
+  in_addr ipv4_bytes;
```

Comment:
Please add a block for ipv6.

The sidecar will run on raw VMs, some on prem using ipv6 - we can't add new code that doesn't support ipv6.

Reply:
https://github.com/istio/proxy/pull/503